packages/components/src/styles/components/application-state.scss

```diff
@@ -7,43 +7,85 @@
 // APPLICATION-STATE COMPONENT
 //
 
-$hds-application-state-padding: 12px 0;
+$hds-application-state-content-max-width: 480px;
 
 .hds-application-state {
-  width: 19.5rem;
-  max-width: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: start;
+  width: fit-content;
   margin: 0 auto; // this will center the component in the parent container
+
+  &.hds-application-state--align-center {
+    align-items: center;
+    text-align: center;
+
+    .hds-application-state__header,
+    .hds-application-state__body,
+    .hds-application-state__footer {
+      width: auto;
+    }
+  }
+}
+
+.hds-application-state__media {
+  margin-bottom: 32px;
+
+  > * {
+    display: block;
+    max-width: 100%;
+  }
 }
 
 .hds-application-state__header {
   display: grid;
+  grid-template-rows: auto auto;
   grid-template-columns: min-content 1fr;
-  align-items: start;
-  color: var(--token-color-foreground-faint);
+  align-items: center;
+  width: 100%;
+  max-width: $hds-application-state-content-max-width;
+  color: var(--token-color-foreground-strong);
+}
+
+.hds-application-state__error-code {
+  grid-column: 1 / -1;
 }
 
 .hds-application-state__icon {
+  grid-row: 2;
   margin-right: 8px; // instead of column gap
   padding-top: 4px; // this seems to align the icon along with line 21
 }
 
-.hds-application-state__title,
-.hds-application-state__error-code {
-  grid-column-start: 2;
+.hds-application-state__title {
+  grid-row: 2;
+  font-weight: var(--token-typography-font-weight-semibold);
+
+  &:not(.hds-application-state__icon + &) {
+    grid-column: 1 / -1;
+  }
 }
 
 .hds-application-state__body {
-  padding: $hds-application-state-padding;
-  color: var(--token-color-foreground-faint);
+  width: 100%;
+  max-width: $hds-application-state-content-max-width;
+  margin: 12px 0 24px;
+  color: var(--token-color-foreground-primary);
 }
 
 .hds-application-state__footer {
   display: flex;
-  gap: 8px;
-  justify-content: space-between;
+  gap: 16px;
+  width: 100%;
+  max-width: $hds-application-state-content-max-width;
+  padding: 12px 0 0;
```

Comment:
Just noticed this after I did my review (not sure how to add to it unless starting a new review?): This padding should be removed, since the in the body is enough.